### ISSUE TITLE
CA1030 for the simplest EventHandler call

### ISSUE THREAD

**Issue by znakeeye** [dotnet/roslyn-analyzers#7623]
The `RaiseFooChanged()` method below is the correct way to raise the `FooChanged` event. Following the DRY principle and all. Still, the analyzer complains about it, and gives a weird suggestion.

> Consider making 'RaiseFooChanged' an event
```c#
public class Foo
{
    public event EventHandler<EventArgs>? FooChanged;

    protected void RaiseFooChanged() => FooChanged?.Invoke(this, EventArgs.Empty);
}
```
The suggestion simply cannot be according to guidelines. If anything, the rule should ask for **the opposite**. If you forget to add a `RaiseSomeEvent()` method, it should ask you to add one, keeping your code clean.
